python/letsql/backends/let/__init__.py

```diff
@@ -242,12 +244,28 @@ def to_pyarrow_batches(
         chunk_size: int = 1_000_000,
         **kwargs: Any,
     ) -> pa.ipc.RecordBatchReader:
-        with self._get_backend_and_expr(expr) as resource:
-            backend, expr = resource
-            return backend.to_pyarrow_batches(expr, chunk_size=chunk_size, **kwargs)
+        expr = self._transform_to_native_backend(expr)
+        expr, created = self._register_and_transform_remote_tables(expr)
+        expr = self._register_and_transform_cache_tables(expr)
+        expr, dt_to_read = self._transform_deferred_reads(expr)
+        backend = self._get_source(expr)
+        if isinstance(backend, self.__class__):
+            backend = super(self.__class__, backend)
+        reader = backend.to_pyarrow_batches(
+            expr.unbind(), chunk_size=chunk_size, **kwargs
+        )
+
+        def clean_up():
+            for table, con in created.items():
+                try:
+                    con.drop_table(table)
+                except Exception:
+                    con.drop_view(table)
+
+        return rbr_wrapper(reader, clean_up)
```

Comment:
can we have `execute` and `to_pyarrow` use `to_pyarrow_batches`?

Reply:
if we do that, we can remove `_get_backend_and_expr`

Reply:
Agreed